src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapper.java

```diff
@@ -125,6 +133,40 @@ public int getNumber() throws AbortException {
         return build().getNumber();
     }
 
+    @Whitelisted
+    public JSONArray getBuildCauses() throws IOException {
+        CauseAction causeAction = build().getAction(CauseAction.class);
+
+        if (causeAction != null) {
+            StringWriter w = new StringWriter();
+            DataWriter writer = JSON.createDataWriter(causeAction, w);
+            Model<CauseAction> model = new ModelBuilder().get(CauseAction.class);
+            model.writeTo(causeAction, writer);
+                JSONObject result =  JSONObject.fromObject(w.toString());
+                // return a slightlly cleaner object
+                return result.getJSONArray("causes");
+        } else {
+            return new JSONArray();
+        }
+    }
+
+    @Whitelisted
```

Comment:
should document what the clazz parameter does.
ie it filters to return only causes that are assignable to type `clazz`  (can this be marked as beta in the help somehow to allow people to actually have somethign now but can be done better latter?)